tensorflow/contrib/rnn/python/ops/rnn_cell.py

```diff
@@ -3392,3 +3392,252 @@ def call(self, inputs, state):
+          of the NTM controller outputs to.
+          If None, no linear projection is applied
+        clip_value: float, The maximum absolute value the controller parameters
+          are clipped to
+        dtype: Default dtype of the layer (default of `None` means use the type
+          of the first input). Required when `build` is called before `call`.
+        reuse: (optional) Python boolean describing whether to reuse variables
+          in an existing scope.  If not `True`, and the existing scope already
+          has the given variables, an error is raised.
+        name: String, the name of the layer. Layers with the same name will
+          share weights, but to avoid mistakes we require reuse=True in such
+          cases.
+    """
+    super(NTMCell, self).__init__(_reuse=reuse, dtype=dtype, name=name)
+
+    rnn_cell_impl.assert_like_rnncell("NTM RNN controller cell", controller)
+
+    self.controller = controller
+    self.memory_size = memory_size
+    self.memory_vector_dim = memory_vector_dim
+    self.read_head_num = read_head_num
+    self.write_head_num = write_head_num
+    self.clip_value = clip_value
+    self.reuse = reuse
+
+    self.NTMControllerState = collections.namedtuple('NTMControllerState',
+      ('controller_state', 'read_vector_list', 'w_list', 'M'))
+
+    self.output_dim = output_dim
+    self.shift_range = shift_range
+
+    self.num_parameters_per_head = self.memory_vector_dim + \
+      2 * self.shift_range + 4
+    self.num_heads = self.read_head_num + self.write_head_num
+    self.total_parameter_num = self.num_parameters_per_head * self.num_heads + \
+      self.memory_vector_dim * 2 * self.write_head_num
+
+  @property
+  def state_size(self):
+    return self.NTMControllerState(
+      controller_state=self.controller.state_size,
+      read_vector_list=[self.memory_vector_dim
+        for _ in range(self.read_head_num)],
+      w_list=[self.memory_size
+        for _ in range(self.read_head_num + self.write_head_num)],
+      M=tensor_shape.TensorShape([self.memory_size * self.memory_vector_dim]))
+
+  @property
+  def output_size(self):
+    return self.output_dim
+
+  def build(self, inputs_shape):
+    if self.output_dim is None:
+      if inputs_shape[1].value is None:
+        raise ValueError("Expected inputs.shape[-1] to be known, saw shape: %s"
+                         % inputs_shape)
+      else:
+        self.output_dim = inputs_shape[1].value
+
+    self._ntm_params_kernel = self.add_variable(
+        'ntm_parameters_kernel',
+        shape=[self.controller.output_size, self.total_parameter_num])
+
+    self._ntm_params_bias = self.add_variable(
+        'ntm_parameters_bias',
+        shape=[self.total_parameter_num],
+        initializer=init_ops.zeros_initializer(dtype=self.dtype))
+
+    self._output_kernel = self.add_variable(
+        'ntm_output_kernel',
+        shape=[self.controller.output_size + \
```

Comment:
avoid backslashes.  in fact, i dont think you need it here since the brackets enclose the computation just fine.  you just need to make sure the indentation is fine on the next line.